lib/charon/token_factory/jwt.ex

```diff
@@ -161,7 +161,7 @@ defmodule Charon.TokenFactory.Jwt do
     %{get_keyset: get_keyset, signing_key: kid} = get_mod_config(config)
 
     with {:ok, key = {alg, _secret}} <- config |> get_keyset.() |> get_key(kid),
-         {:ok, json_payload} <- jmod.encode(payload) do
```

Comment:
JSON only has `encode!/1` and there's really no reason to use `encode/1` here, since it would be a major programming error to have non-serializable input here.